### ISSUE TITLE
`prefer-top-level-await`: Miss cases

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1502]
Currently, it only reports top-level expressions, the following cases should be reported too.

```js
if (foo) {
	bar().catch(() => {});
}
```

```js
const promise = Promise.resolve();
for (const foo of bar) {
	promise.then(foo);
}
```

I'll work on it, when I have time.
